chain/types/ethtypes/rlp.go

```diff
@@ -134,9 +134,12 @@ func decodeRLP(data []byte) (res interface{}, consumed int, err error) {
 			return nil, 0, err
 		}
 		totalLen := 1 + strLenInBytes + strLen
-		if totalLen > len(data) {
+		if totalLen > len(data) || totalLen < 0 {
```

Comment:
@fridrik01 Isn't that the check you already added in https://github.com/filecoin-project/lotus/pull/11079? Sorry, I'm a little confused.

In this PR I'm trying to catch an edge-case where `decodedLength` doesn't overflow, but `decodedLength + strLenInBytes + 1` does.

Reply:
Ah, got it! I think I'm gonna do both -- your suggestion SGTM, but we do have to check again in the caller because we wind up indexing `data[totalLen + 1]`